api/src/integration/java/org/pmiops/workbench/google/DirectoryServiceImplIntegrationTest.java

```diff
@@ -41,10 +44,12 @@ public void testDirectoryServiceUsernameIsTaken() {
   @Test
   public void testCreateAndDeleteTestUser() {
     String userName = String.format("integration.test.%d", Clock.systemUTC().millis());
-    service.createUser("Integration", "Test", userName, "notasecret");
-    assertThat(service.isUsernameTaken(userName)).isTrue();
-    service.deleteUser(userName);
-    assertThat(service.isUsernameTaken(userName)).isFalse();
+    DirectoryServiceImpl serviceSpy = Mockito.spy(service);
```

Comment:
This works... but I'd recommend having a separate service for sending email injected into DirectoryServiceImpl. Then you can just @mock that service out normally here (or anywhere else we send email) instead.

Reply:
I like that idea too. We're in a bit of a time crunch for the release, however, so I don't want to implement new code without solid testing. I will make a new branch with this idea and try to push that through post-release.